packages/blockchain-wallet-v4-frontend/src/store/index.js

```diff
@@ -37,12 +36,23 @@ const configuredStore = async function () {
   // immediately load app configuration
   let options
   try {
-    let res = await fetch('/wallet-options-v4.json')
+    const res = await fetch('/wallet-options-v4.json')
     options = await res.json()
   } catch (e) {
     throw new Error('errorWalletOptionsApi')
   }
 
+  // generate xSessionId and store in local storage
+  try {
+    const res = await axios.put(`${options.domains.api}/nabu-gateway/generate-session`)
+    const { xSessionId } = res.data
+    if (xSessionId) {
+      localStorage.setItem(model.profile.X_SESSION_ID, xSessionId)
+    }
+  } catch (e) {
+    throw new Error('errorXSessionId')
```

Comment:
Are you sure you want to `throw` here? I think it will stop the entire app from loading.